examples/notebooks/load-modelling.ipynb

```diff
@@ -0,0 +1,508 @@
+    "plt.xlim([0, 0.5])\n",
+    "plt.ylim([0, 150])\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the same network as in the previous sections, we perform the iteration:\n",
+    "- add demand to network\n",
+    "- optimize network to get the system price\n",
+    "- update the demand by adding the system price to the elasticity curve"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "base_load = 0.2\n",
+    "all_networks = []\n",
+    "n = network\n",
+    "\n",
+    "for _ in range(6):\n",
+    "    n = n.copy()\n",
+    "    load = pd.Series(\n",
+    "        base_load, index=range(24)\n",
+    "    )  # variable baseload based on base_load_values\n",
+    "    n.remove(\"Load\", \"load\")\n",
+    "    n.add(\"Load\", \"load\", bus=\"My bus\", p_set=load.values)\n",
+    "    n.lopf()\n",
+    "\n",
+    "    # Estimated price of the network in euros per MWh\n",
+    "    system_cost_per_MWh = (n.objective + n.objective_constant) / n.loads_t.p.sum().sum()\n",
+    "\n",
+    "    # assuming a linear demand response to price\n",
+    "    new_load = elasticity(system_cost_per_MWh)\n",
+    "    base_load = new_load\n",
+    "\n",
+    "    all_networks.append(n)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Let's plot the elastic demand"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Demand and price values from the created networks\n",
+    "average_loads = []\n",
+    "prices = []\n",
+    "for n in all_networks:\n",
+    "    average_loads.append(n.loads_t.p.mean().sum())\n",
+    "    prices.append((n.objective + n.objective_constant) / n.loads_t.p.sum().sum())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_demand_iteration = average_loads[0]\n",
+    "updated_average_loads = average_loads[\n",
+    "    1:\n",
+    "]  # demand value in first iteration didn't have a price\n",
+    "updated_prices = prices[:-1]  # last price value wasn't used\n",
+    "\n",
+    "# Scatter plot of the points\n",
+    "plt.scatter(updated_average_loads, updated_prices, label=\"Elastic demand\")\n",
+    "\n",
+    "# Add numbers to each point\n",
+    "for i, (x, y) in enumerate(zip(updated_average_loads, updated_prices)):\n",
+    "    plt.annotate(\n",
+    "        \"Iteration \" + str(i + 1),\n",
+    "        (x, y),\n",
+    "        textcoords=\"offset points\",\n",
+    "        xytext=(30, 5),\n",
+    "        ha=\"center\",\n",
+    "    )\n",
+    "\n",
+    "# Fit a linear curve to the data\n",
+    "slope, intercept = np.polyfit(updated_average_loads, updated_prices, 1)\n",
```

Comment:
Could it probably make sense to add overlay results of the simulation with the original elasticity curve? @AnasAlgarei what is your feeling about that?

Reply:
The original line is already overlayed, but it is zoomed in the graph for the points to be clear to see.
![image](https://github.com/PyPSA/PyPSA/assets/101210563/3599416e-eac6-4427-88ad-8073a197dfbb)

When zooming out the points will be very close to each other that it is not clear:
![image](https://github.com/PyPSA/PyPSA/assets/101210563/b2926110-9b1d-420c-bc2e-7ebcaae63a99)

Reply:
@AnasAlgarei absolutely agree regarding zooming :) An attempt to keep the original scale doesn't look nice.

I have been thinking about making more evident that the elasticity curve above and here are the same thing, like continuing the elasticity line beyond the modeled points and probably trying to keep the same angle with the axises. But that would require additions to the code, which would be distractive. So, let's keep as is. 

I think, the legend will be enough to make the connection between both plots.